Description: Exclude MRI tests that are consistently failing
 Upstream runs most of their tests in openjdk-7, but the default
 JDK in Debian currently is openjdk-8, so this difference could
 cause some failures. In any case, investigation is needed.
Author: Miguel Landaeta <nomadium@debian.org>
Forwarded: no
Last-Update: 2017-03-11

--- /dev/null
+++ jruby-9.1.8.0/test/mri/excludes/TestCoverage.rb
@@ -0,0 +1,3 @@
+exclude :test_coverage_snapshot, "needs investigation"
+exclude :test_peek_result_without_start, "needs investigation"
+exclude :test_restarting_coverage, "needs investigation"
--- jruby-9.1.8.0.orig/test/mri/excludes/TestEval.rb
+++ jruby-9.1.8.0/test/mri/excludes/TestEval.rb
@@ -1,3 +1,4 @@
 exclude :test_eval_ascii_incompatible, "needs investigation"
 exclude :test_eval_location_fstring, "needs investigation"
 exclude :test_file_encoding, "needs investigation"
+exclude :test_eval_with_toplevel_binding, "needs investigation"
--- /dev/null
+++ jruby-9.1.8.0/test/mri/excludes/TestJSONFixtures.rb
@@ -0,0 +1 @@
+exclude :test_failing, "needs investigation"
--- jruby-9.1.8.0.orig/test/mri/excludes/TestJSONGenerate.rb
+++ jruby-9.1.8.0/test/mri/excludes/TestJSONGenerate.rb
@@ -3,3 +3,10 @@ exclude :test_gc, "needs investigation"
 exclude :test_hash_likeness_set_string, "needs investigation"
 exclude :test_hash_likeness_set_symbol, "needs investigation"
 exclude :test_string_subclass, "hard call of json blows up; probably fixed in C ext. https://github.com/flori/json/issues/272"
+exclude :test_fast_generate, "needs investigation"
+exclude :test_fast_state, "needs investigation"
+exclude :test_generate, "needs investigation"
+exclude :test_generate_pretty, "needs investigation"
+exclude :test_own_state, "needs investigation"
+exclude :test_pretty_state, "needs investigation"
+exclude :test_safe_state, "needs investigation"
--- jruby-9.1.8.0.orig/test/mri/excludes/TestMethod.rb
+++ jruby-9.1.8.0/test/mri/excludes/TestMethod.rb
@@ -14,3 +14,4 @@ exclude :test_splat_long_array, "passes
 exclude :test_super_method_removed, "finds super method when super method has been undef (#2155)"
 exclude :test_to_proc_binding, "NullPointerException in parser"
 exclude :test_visibility, "fails in compiler"
+exclude :test_callee_top_level, "needs investigation"
--- jruby-9.1.8.0/test/mri/excludes/TestProcess.rb
+++ jruby-9.1.8.0/test/mri/excludes/TestProcess.rb
@@ -67,3 +67,4 @@ exclude :test_system_wordsplit, "needs investigation"
 exclude :test_uid_re_exchangeable_p, "unimplemented"
 exclude :test_uid_sid_available?, "unimplemented"
 exclude :test_wait_exception, "Interrupt expected but nothing was raised."
+exclude :test_popen_exit, "needs investigation, it gets killed"
--- jruby-9.1.8.0/test/mri/excludes/TestDefined.rb
+++ jruby-9.1.8.0/test/mri/excludes/TestDefined.rb
@@ -1 +1,2 @@
 exclude :test_defined, "needs investigation"
+exclude :test_defined_impl_specific, "flaky, needs investigation"
--- jruby-9.1.8.0/test/mri/excludes/TestModule.rb
+++ jruby-9.1.8.0/test/mri/excludes/TestModule.rb
@@ -14,3 +14,6 @@ exclude :test_private_constant_with_no_args, "needs investigation"
 exclude :test_protected_singleton_method, "needs investigation"
 exclude :test_undef, "needs investigation"
 exclude :test_prepend_each_classes, "needs investigation - ancestors do not match"
+exclude :test_ancestors, "needs investigation"
+exclude :test_included_module, "needs investigation"
+exclude :test_included_modules, "needs investigation"
--- jruby-9.1.8.0/test/mri/excludes/TestFileExhaustive.rb
+++ jruby-9.1.8.0/test/mri/excludes/TestFileExhaustive.rb
@@ -7,4 +7,5 @@ exclude :test_expand_path_returns_tainted_strings_or_not, "needs investigation"
 exclude :test_extname, "needs investigation"
 exclude :test_join_ascii_incompatible, "needs investigation"
 exclude :test_lchmod, "Fails on Linux"
-exclude :test_readlink_long_path, "needs investigation"
\ No newline at end of file
+exclude :test_readlink_long_path, "needs investigation"
+exclude :test_expand_path_for_existent_username, "sbuild breaks this test since it modifies the environment"
